VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -975,15 +966,19 @@ private boolean isScreenLandscape() {
         @Override
         public void run() {
             if (isResumed() && !isRemoving()) {
-                hideProgress();
-                if(player!=null) {
+                if (player != null) {
                     player.unfreeze();
+                    hideProgress();
                     if (player.isPlaying()) {
-                        hideProgress();
                         updateController("player unfreezed");
                     }
 
-                    if (pauseDueToDialog){
+                    if (IS_AUTOPLAY_ENABLED && isPrepared && !isAutoPlayDone) {
+                        isAutoPlayDone = true;
+                        player.start();
```

Comment:
OK, so I guess there is no way to change the video state without unfreezing it. I guess the current implementation is fine then. There should also be a check for `getUserVisibleHint()` at the start of the callback though, just like in `onPrepared()` (the callback is removed is `onPause()`, but `setUserVisibleHint()` is not called at all for child fragments so there is no way to have it removed from the event itself). Then this case should be handled completely.